ext/bootsnap/bootsnap.c

```diff
@@ -713,8 +712,8 @@ bs_fetch(char * path, VALUE path_v, char * cache_path, VALUE handler, VALUE args
     else if (res == CACHE_UNCOMPILABLE) {
       /* If fetch_cached_data returned `Uncompilable` we fallback to `input_to_output`
         This happens if we have say, an unsafe YAML cache, but try to load it in safe mode */
-      if (bs_read_contents(current_fd, current_key.size, &contents, &errno_provenance) < 0) goto fail_errno;
-      input_data = rb_str_new(contents, current_key.size);
+      input_data = rb_str_buf_new(cached_key.size);
+      if (bs_read_contents(current_fd, current_key.size, input_data, &errno_provenance) < 0) goto fail_errno;
```

Comment:
Since the return value of `bs_read_contents` is always checked against `< 0`, you could move `rb_str_buf_new` inside, and return `Qfalse` on failure.

Reply:
Ok, made those adjustments in https://github.com/Shopify/bootsnap/pull/437/commits/d1c01f6a6f83d2cd307624be5085eecac3d8b5ad